### PR TITLE
Add all positive cases for GlassFish runtime 

### PR DESCRIPTION
**Pull request by adan-jawad**

### CODE REVIEW

**Review by m-reza-rahman — CHANGES_REQUESTED** [review]
This is organized incorrectly. At the top level, there should be two jobs: one for the Archetype and one for the UI. For the Archetype, things should first be organized by Java SE version, corresponding to invocations with that version. For each Java SE version, tests should then be organized by runtime, Jakarta EE version, profile then Docker flag.

**Comment by adan-jawad** [comment]
Hello, I have added descriptions for docker support and organized the cases in the way specified. I have restored the old cases from the original file and have added my cases between the start and end comments for every java version for easy readability. I'll remove these comments once you review this.

**Review by m-reza-rahman — CHANGES_REQUESTED** [review]
Please order by SE version (starting with the lowest), runtime/none (alphabetic, starting with none), EE version (starting with the lowest), profile (full, web, core), and Docker (no, yes).

So like this:

* Run Archetype for EE 8 Web Profile, SE 8, Payara
* Run Archetype for EE 8 Web Profile, SE 8, Payara, with Docker

Where Docker is really not supported (a warning is generated by the Archetype), we should not treat that as a "positive case". That's a negative case we need to test for later. GlassFish currently does not support Docker.

Please order your part and the file overall - including existing cases. Confusing to read and evaluate otherwise for me.

**Comment by m-reza-rahman** [comment]
Please resolve conflicts with latest commits to master.

**Comment by adan-jawad** [comment]
I am a bit confused. Do I delete the cases where docker is not supported, or do I just remove that from the description?

**Comment by m-reza-rahman** [comment]
For now, kindly remove the cases where Docker is not supported and set them aside. These need to be added later as explicitly tested negative cases.

**Comment by m-reza-rahman** [comment]
Please resolve all merge conflicts.

**Comment by m-reza-rahman** [comment]
I will review if I have some time during the work week. Otherwise, may need to wait until the weekend.

**Comment by adan-jawad** [comment]
> I will review if I have some time during the work week. Otherwise, may need to wait until the weekend.

Sure.

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -34,6 +34,9 @@ jobs:
       - name: Run Archetype for EE 8 Web Profile, SE 8
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=8 -Dprofile=web -DjavaVersion=8 -DoutputDirectory=app -Dgoals="clean package"
+
+          rm -rf app
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:38]: Why is this needed?

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -56,11 +59,18 @@ jobs:
...
           rm -rf app
 
+
       - name: Run Archetype for EE 8, SE 8, GlassFish
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=8 -Dprofile=full -DjavaVersion=8 -Druntime=glassfish -DoutputDirectory=app/glassfish -Dgoals="clean package"
           rm -rf app/glassfish
 
+
+      - name: Run Archetype for EE 8 Web Profile, SE 8, GlassFish
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:69]: This is a duplicate. Why is it needed?

File: .github/workflows/nightly.yml [hunk 3]
```diff
@@ -133,6 +144,10 @@ jobs:
 
       - name: Run Archetype for EE 8, SE 8, WildFly
         run: |
+
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:147]: Why was this added?

File: .github/workflows/nightly.yml [hunk 4]
```diff
@@ -503,13 +602,18 @@ jobs:
...
 
+
       - name: Run Archetype for EE 10 Web Profile, SE 17, WildFly
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=10 -Dprofile=web -DjavaVersion=17 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
           rm -rf app/wildfly
 
       - name: Run Archetype for EE 8, SE 17, WildFly
         run: |
+
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:613]: Why was this added?

File: .github/workflows/nightly.yml [hunk 5]
```diff
@@ -151,12 +166,54 @@ jobs:
...
           rm -rf app/wildfly
 
+
       - name: Set up Java SE 11
         uses: actions/setup-java@v3
         with:
           distribution: "temurin"
           java-version: 11
 
+
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:176]: Where are the SE 8 cases for GlassFish?

**Review by m-reza-rahman — CHANGES_REQUESTED** [review]
My guess is that this PR has some improper merge conflict resolution issues. Please re-inspect and fix.

File: .github/workflows/nightly.yml [hunk 6]
```diff
@@ -9,23 +9,23 @@ jobs:
   build-archetype:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:12]: Please pull the latest from master. This diff should disappear.

File: .github/workflows/nightly.yml [hunk 7]
```diff
@@ -9,23 +9,23 @@ jobs:
...
       - name: Set up Java SE 8
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
         with:
           distribution: "temurin"
           java-version: 8
 
       - name: Cache Maven packages
-        uses: actions/cache@v3
+        uses: actions/cache@v4
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:20]: Please pull the latest from master. This diff should disappear.

File: .github/workflows/nightly.yml [hunk 8]
```diff
@@ -149,10 +151,12 @@ jobs:
       - name: Run Archetype for EE 8 Web Profile, SE 8, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=8 -Dprofile=web -DjavaVersion=8 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
+          docker rmi test-image
           rm -rf app/wildfly
 
       - name: Set up Java SE 11
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:159]: Please pull the latest from master. This diff should disappear.

File: .github/workflows/nightly.yml [hunk 9]
```diff
@@ -350,10 +402,12 @@ jobs:
       - name: Run Archetype for EE 10 Core Profile, SE 11, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=10 -Dprofile=core -DjavaVersion=11 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
+          docker rmi test-image
           rm -rf app/wildfly
 
       - name: Set up Java SE 17
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:410]: Please pull the latest from master. This diff should disappear.

File: .github/workflows/nightly.yml [hunk 10]
```diff
@@ -556,25 +653,26 @@ jobs:
...
+          docker build -t test-image app/wildfly/jakartaee-hello-world
+          docker rmi test-image
           rm -rf app/wildfly
 
-
   build-ui:
     runs-on: ubuntu-20.04
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:663]: Please pull the latest from master. This diff should disappear.

File: .github/workflows/nightly.yml [hunk 11]
```diff
@@ -556,25 +653,26 @@ jobs:
...
       - name: Setup Java
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:665]: Please pull the latest from master. This diff should disappear.

File: .github/workflows/nightly.yml [hunk 12]
```diff
@@ -556,25 +653,26 @@ jobs:
...
         with:
           distribution: "temurin"
           java-version: 11
 
       - name: Cache Maven packages
-        uses: actions/cache@v3
+        uses: actions/cache@v4
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:671]: Please pull the latest from master. This diff should disappear.

File: .github/workflows/nightly.yml [hunk 13]
```diff
@@ -9,23 +9,23 @@ jobs:
...
         with:
           path: ~/.m2
           key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ runner.os }}-m2
 
       - name: Build Archetype
         run: mvn clean install --file archetype/pom.xml
-      
+
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:28]: Why is this white space change needed?

File: .github/workflows/nightly.yml [hunk 14]
```diff
@@ -125,7 +125,7 @@ jobs:
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=9.1 -Dprofile=web -DjavaVersion=8 -Druntime="open-liberty" -Ddocker=yes -DoutputDirectory="app/open-liberty" -Dgoals="clean package"
           rm -rf app/open-liberty
-      
+
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:128]: Why is this white space change needed?

File: .github/workflows/nightly.yml [hunk 15]
```diff
@@ -291,7 +335,7 @@ jobs:
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=10 -Dprofile=core -DjavaVersion=11 -Druntime="open-liberty" -Ddocker=yes -DoutputDirectory="app/open-liberty" -Dgoals="clean package"
           rm -rf app/open-liberty
-      
+
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:338]: Why is this white space change needed?

File: .github/workflows/nightly.yml [hunk 16]
```diff
@@ -497,7 +586,7 @@ jobs:
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=10 -Dprofile=core -DjavaVersion=17 -Druntime="open-liberty" -Ddocker=yes -DoutputDirectory="app/open-liberty" -Dgoals="clean package"
           rm -rf app/open-liberty
-  
+
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:589]: Why is this white space change needed?

File: .github/workflows/nightly.yml [hunk 17]
```diff
@@ -556,25 +653,26 @@ jobs:
...
+        uses: actions/cache@v4
         with:
           path: ~/.m2
           key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ runner.os }}-m2
 
       - name: Build UI
-        run: mvn clean package --file ui/pom.xml
\ No newline at end of file
+        run: mvn clean package --file ui/pom.xml
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:678]: Why is this white space change needed?

File: .github/workflows/nightly.yml [hunk 18]
```diff
@@ -139,6 +139,8 @@ jobs:
       - name: Run Archetype for EE 8, SE 8, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=8 -Dprofile=full -DjavaVersion=8 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:142]: Please remove this. This is not related to GlassFish.

File: .github/workflows/nightly.yml [hunk 19]
```diff
@@ -9,23 +9,23 @@ jobs:
   build-archetype:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - name: Set up Java SE 8
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:14]: Please pull the latest from master. This diff should disappear.

File: .github/workflows/nightly.yml [hunk 20]
```diff
@@ -149,10 +151,12 @@ jobs:
       - name: Run Archetype for EE 8 Web Profile, SE 8, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=8 -Dprofile=web -DjavaVersion=8 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:154]: Please remove this. This is not related to GlassFish.

File: .github/workflows/nightly.yml [hunk 21]
```diff
@@ -310,6 +354,8 @@ jobs:
       - name: Run Archetype for EE 8, SE 11, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=8 -Dprofile=full -DjavaVersion=11 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:357]: Please remove this. This is not related to GlassFish.

File: .github/workflows/nightly.yml [hunk 22]
```diff
@@ -320,6 +366,8 @@ jobs:
       - name: Run Archetype for EE 8 Web Profile, SE 11, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=8 -Dprofile=web -DjavaVersion=11 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:369]: Please remove this. This is not related to GlassFish.

File: .github/workflows/nightly.yml [hunk 23]
```diff
@@ -330,6 +378,8 @@ jobs:
       - name: Run Archetype for EE 10, SE 11, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=10 -Dprofile=full -DjavaVersion=11 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:381]: Please remove this. This is not related to GlassFish.

File: .github/workflows/nightly.yml [hunk 24]
```diff
@@ -340,6 +390,8 @@ jobs:
       - name: Run Archetype for EE 10 Web Profile, SE 11, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=10 -Dprofile=web -DjavaVersion=11 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:393]: Please remove this. This is not related to GlassFish.

File: .github/workflows/nightly.yml [hunk 25]
```diff
@@ -350,10 +402,12 @@ jobs:
       - name: Run Archetype for EE 10 Core Profile, SE 11, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=10 -Dprofile=core -DjavaVersion=11 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:405]: Please remove this. This is not related to GlassFish.

File: .github/workflows/nightly.yml [hunk 26]
```diff
@@ -516,6 +605,8 @@ jobs:
       - name: Run Archetype for EE 8, SE 17, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=8 -Dprofile=full -DjavaVersion=17 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:608]: Please remove this. This is not related to GlassFish.

File: .github/workflows/nightly.yml [hunk 27]
```diff
@@ -526,6 +617,8 @@ jobs:
       - name: Run Archetype for EE 8 Web Profile, SE 17, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=8 -Dprofile=web -DjavaVersion=17 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:620]: Please remove this. This is not related to GlassFish.

File: .github/workflows/nightly.yml [hunk 28]
```diff
@@ -536,6 +629,8 @@ jobs:
       - name: Run Archetype for EE 10, SE 17, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=10 -Dprofile=full -DjavaVersion=17 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:632]: Please remove this. This is not related to GlassFish.

File: .github/workflows/nightly.yml [hunk 29]
```diff
@@ -546,6 +641,8 @@ jobs:
       - name: Run Archetype for EE 10 Web Profile, SE 17, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=10 -Dprofile=web -DjavaVersion=17 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:644]: Please remove this. This is not related to GlassFish.

File: .github/workflows/nightly.yml [hunk 30]
```diff
@@ -556,25 +653,26 @@ jobs:
       - name: Run Archetype for EE 10 Core Profile, SE 17, WildFly, with Docker
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=10 -Dprofile=core -DjavaVersion=17 -Druntime=wildfly -Ddocker=yes -DoutputDirectory=app/wildfly -Dgoals="clean package"
+          docker build -t test-image app/wildfly/jakartaee-hello-world
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:656]: Please remove this. This is not related to GlassFish.

**Review by m-reza-rahman — CHANGES_REQUESTED** [review]
Please address specific in-place feedback. Then I can review hopefully one last time before merge.

**Comment by adan-jawad** [comment]
Hello Reza, I am going to create a new pull request and I'm closing this one
